R/prepare_geo_data.R

```diff
@@ -52,23 +54,90 @@ prepare_geo_data <- function(data,
     inner_join(nuts_de, by = "geo")
 
   # merge shapefile with financial data
-  financial_geo <- data |>
+  geo <- data |>
     filter(benchmark == benchmark_arg) |>
     left_join(shp_1, by = "postcode") |>
     st_as_sf()
 
-  # Add the code for data aggregation and color mapping
-  aggregated_data <- financial_geo |>
-    group_by(.data$postcode, .data$xctr_risk_category) |>
-    summarize(count = n()) |>
-    group_by(.data$postcode) |>
-    mutate(proportion = .data$count / sum(.data$count)) |>
-    ungroup()
+  aggregated_data <- aggregate_geo_data(geo, mode)
+
+  return(list(shp_1, aggregated_data))
```

Comment:
> Only use `return()` for early returns. Otherwise, rely on R to return the result of the last evaluated expression.

https://style.tidyverse.org/functions.html?q=return#return